juju/juju
# Store caas container info in separate collection; optimise status reporting

## Description of change

CAAS container information is now stored in a separate collection instead of as attributes on the unit.
This is needed so that unit is less polluted and we can watch for container changes.
Also report in status when a pod is terminated by the caas substrate, and optimise status reporting by not sending a substrate reported status value unless it has changed.

## QA steps

Deploy a CAAS cloud and deploy the mysql charm, check status.
Wait for k8s to decide it wants to restart the pod and ensure this is reflected in juju status.
